pkg/types/types.go

```diff
@@ -184,6 +195,24 @@ type DeviceProvider interface {
 	ValidConfig(*ResourceConfig) bool
 }
 
+// Device interfaces relationship:
```

Comment:
please remove this ascii drawing from source code as its clear from the interfaces below who embeds who.
also if we keep it we need to maintain it.